common/lib/xmodule/xmodule/js/src/video/03_video_player.js

```diff
@@ -805,9 +797,11 @@ function (HTML5Video, Resizer) {
     }
 
     function updatePlayTime(time, skip_seek) {
-        var videoPlayer = this.videoPlayer,
-            duration = this.videoPlayer.duration(),
-            youTubeId;
+        var videoPlayer = this.videoPlayer;
+        var duration = (this.config.endTime !== null) ?
```

Comment:
We can get a smaller diff here with:

```javascript
// This block is unchanged.
var videoPlayer = this.videoPlayer,
    duration = this.videoPlayer.duration(),
    youTubeId;

// This is the only part that changes.
if (this.config.endTime !== null) {
    duration = Math.min(this.config.endTime, duration);
```

This way, the only change here is adding the small `if` statement without deleting any code (a purely additive diff).